### DISCUSSION TITLE
Scheduling Defender for macOS Antivirus Scans in Intune - odds+endpoints

### DISCUSSION THREAD

**Discussion by giscus** [ennnbeee/oddsandendpoints-web#9] [Comments]
#  Scheduling Defender for macOS Antivirus Scans in Intune 

Using PowerShell to create Defender antivirus scan setting custom profiles for Microsoft Intune and third-party MDM enrolled macOS devices.

https://memv.ennbee.uk/posts/macos-defender-scan-settings/

**Comment by lefrinj** [comment]
Hi, thanks for making and sharing this script. I found you from Andrew Taylor's Intune newsletter. When I'm trying to upload a mobileconfig file, it tells me 'Must be a well formed XML string' - but I've put it through an XML validator and also tried pasting in one of your examples from above, so I know it is valid. I'll try again sometime in a few days in case Intune is just having a bad day...

**Reply by ennnbeee** [reply]
Let me know how you get on, I'll test this again today to make sure that it's working as expected in a few tenants I have access to.

**Reply by ennnbeee** [reply]
Ahh ok, @lefrinj so it's the file itself that's the issue not the content. When copying the content out of the generated file to a new file, Intune is happy with it, but not the file the script creates.

I imagine it's a UTF-8 file format issue, I'll look at updating the script and let you know when complete.

**Reply by ennnbeee** [reply]
@lefrinj script has been updated, just needed to include `-Encoding utf8` when exporting the XML file.

**Comment by lefrinj** [comment]
Oo nice spot, I didn't think of that. Thanks very much for updating me (and so quickly)

Dev

________________________________
